### PR TITLE
Created a SuperTextFieldContext with a BuildContext and a TextFieldScroller - passed it key handlers (Resolves #1487)

### LINKED ISSUE #1487: [SuperTextField] Introduce a scrolling api
**Description:**

Currently, `SuperTextField` doesn't have a proper scrolling api which exposes its scrolling artefacts. Due to the absence of such api, we don't have a single source that could alter the scrolling of the `SuperTextField` content and we've to rely on manually passing the scrolling artefacts through constructors at the places they're needed. This is resulting in boilerplate code that could be avoided by addition of an scrolling api that exposes this artefacts wherever needed. 

Compared to `SuperTextField`, `SuperEditor` has its own scrolling api that exposes the scrolling artefacts and scroll actions on `SuperEditor` through `SuperEditorContext`.  We could try to introduce something like this for `SuperTextField`. 

This scrolling api could prove helpful in case of `SuperDesktopField` where we've to scroll the textfield as well as the ancestor document when scroll actions are performed on certain keyboard shortcuts( #1463  ). Currently, we're passing the scrolling artefacts created within `SuperDesktopField` that manages its scrolling, directly to the scroll key handlers which has resulted in addition of boilerplate code that could be avoided by such api.

**Api requirements:**

I'm not sure abt the full specs of this api but some off the top of my mind are,

1. Access to `ScrollController` that controls textfield: This could be tricky. Currently there's no particular scroll controller within `SuperTextField` for this. But rather, `SuperDesktopField`, `SuperAndroidTextField` and `SuperIOSTextField` has their own independent ways to manage the scrolling of textfield within them which is not known to `SuperTextField`.

2. Access to ancestor scrollable if one's present.
3. Able to scroll content within the `SuperTextField`.
4. Able to scroll the ancestor scrollable if one's present.

### PR DESCRIPTION
**Pull request by matthew-carroll**
Created a SuperTextFieldContext with a BuildContext and a TextFieldScroller - passed it key handlers (Resolves #1487)

### CODE REVIEW

**Comment by matthew-carroll** [comment]
@rutvik110 - please let me know if this API would work for your key handlers that scroll text fields

**Comment by rutvik110** [comment]
This looks really great! I'll give it a try with `SuperDesktopTextField` scroll handlers.

**Comment by rutvik110** [comment]
Just tried this updates on the `SuperDesktopTextField` scroll key handlers, works nicely! 

Beside this, I identified an issue when we are setting `TextInputSource` to `ime` for textfields, an extra line is being added into the textfield on enter. This behaviour is seen across both `SuperTextField` and `SuperDesktopField`. 

https://github.com/superlistapp/super_editor/assets/65209850/66e8c2c3-cabf-4c58-87f1-14e0f93d0eb2

**Comment by angelosilvestre** [comment]
@rutvik110 We have an ongoing PR to fix that issue: https://github.com/superlistapp/super_editor/pull/1469

File: super_editor/lib/src/super_textfield/desktop/desktop_textfield.dart [hunk 1]
```diff
@@ -171,6 +174,10 @@ class SuperDesktopTextFieldState extends State<SuperDesktopTextField> implements
     _controller.addListener(_onSelectionOrContentChange);
 
     _scrollController = ScrollController();
+    _textFieldScroller = TextFieldScroller() //
+      ..attach(_scrollController);
```
Comment (angelosilvestre) [inline super_editor/lib/src/super_textfield/desktop/desktop_textfield.dart:178]: It seems we are not detaching from the scroll controller anywhere.

**Review by angelosilvestre — APPROVED** [review]
LGTM with a single comment.

**Pull request merged**